app/src/authFetch.ts

```diff
@@ -28,9 +28,15 @@ export async function authFetch(
   } catch (error) {
     if (error instanceof UnauthorizedError) {
       // If the server returns a 401, we should try to refresh the token
-      await refreshTokens();
-      // Retry the original request
-      return fetch(input, init);
+      const response = await refreshTokens();
+      // If there is no response, the token was successfully refreshed
+      if (!response) {
+        // Retry the original request
+        return fetch(input, init);
+      }
+      // If there is a response, it is a redirect response, and we should return it
+      // for any caller to handle / follow as needed
+      return response;
```

Comment:
this slightly feels "tribal knowledge" in nature. is there a more declarative way to do this. using response nullishness is hard to explain